### PR TITLE
Enable PluginImageHelper getting image from another plugin.

### PR DESCRIPTION
**Pull request by gallandarakhneorg**
Signed-off-by: Stéphane Galland <galland@arakhne.org>

### CODE REVIEW

**Comment by cdietrich** [comment]
I think it’s a bad idea to use jdt code
Maybe looking at this bug will give a better solution 
https://bugs.eclipse.org/bugs/show_bug.cgi?id=404741

**Comment by kthoms** [comment]
Second that. We had recently trouble with JDT internals, we should avoid it. Could you find a way with public API only?

**Comment by gallandarakhneorg** [comment]
Make sense to me. I will update my PR by removing the JDT reference, and taking into consideration the proposal from Christian.

**Comment by gallandarakhneorg** [comment]
Reference to JDT can be removed safely. I used in a previous version it but not any more.
The code within the bug report provided by Christian seems to be the same as the one already inside `PluginImageHelper`, am I right? It does not consider references to images within another plugin as the current plugin. I have updated the code for enabling cross-plugin references.

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java [hunk 1]
```diff
@@ -184,4 +193,26 @@ public void bundleChanged(BundleEvent event) {
 		}
 	}
 
+	/** Find the descriptor of the image with the given id.
+	 *
+	 * @param name the identifier of the image. It may be qualified with the plugin's id.
+	 * @return the descriptor.
+	 * @since 2.14
+	 */
+	protected ImageDescriptor findImage(String name) {
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: what is the name you are looking for? how do you encode the image name

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: pluginname.imagename.extension?

Comment (gallandarakhneorg) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: The supported format is of name is "<pluginid>.<imagename>".

Comment (cdietrich) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: Can you please make this a bit more explicit in the javadoc with an example

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: I suggest to use this syntax for cross-bundle lookup:
`/bundlename/path/image.name.with.dot.ext`
That is, a leading slash and a bundleName, and slash-delimited path to the resource of that bundle.

Comment (gallandarakhneorg) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: We could also use the standard Eclipse RCP notation: `platform:/plugin/<bundlename>/<path>/<image.name.with.dot.ext>`

What is your preferred one?

Comment (kthoms) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: I‘d prefer platform plugin URIs. 
@szarnekow @cdietrich WDYT?

Comment (szarnekow) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: I don't have a strong preference. Most importantly it has to be unambiguous. Testing for a '/' as the first char and putting `platform:/plugin` in front of it are requiring the complete URI is both ok to me.

Comment (gallandarakhneorg) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: I have use the platform-uri syntax.

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java [hunk 2]
```diff
@@ -23,11 +23,30 @@
...
 
-/**
+/** Helper for loading images from plugins.
+ *
+ * <p>
+ * This helper enables to find the images within the current plugin and other
+ * plugins. In case the image is located within another plugin than the current
+ * one, the image's name should be qualifed with the plugin's identifier:
+ * {@code platform:/plugin/<plugin_identifier>/<image_name>.<image_filename_extension>}.
+ * The image name must not contains a dot character.
```
Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: typo: must not contain...

Comment (gallandarakhneorg) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: Done.

Comment (gallandarakhneorg) [reply org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/PluginImageHelper.java]: done.

**Comment by kthoms** [comment]
Thank you, Stéphane!

**Pull request merged**